sound/soc/sof/pm.c

```diff
@@ -411,6 +411,26 @@ int snd_sof_runtime_resume(struct device *dev)
 }
 EXPORT_SYMBOL(snd_sof_runtime_resume);
 
+int snd_sof_set_d0_substate(struct snd_sof_dev *sdev,
+			    enum sof_d0_substate d0_substate)
+{
+	int ret;
+
+	/* do platform specific set_state */
+	ret = snd_sof_dsp_set_power_state(sdev, d0_substate);
+	if (ret < 0) {
```

Comment:
this needs to change now that you've allowed the -ENOTSUP return value. You should only log an error when the return is not -ENOTSUP.

Reply:
let me just remove logging here, we have check in the caller place already.